src/core/execute.c

```diff
@@ -3596,6 +3596,13 @@ bool exec_context_get_cpu_affinity_from_numa(const ExecContext *c) {
 static int add_shifted_fd(int *fds, size_t fds_size, size_t *n_fds, int fd, int *ret_fd) {
         int r;
 
+        /* Let's move the fd up, so that it's outside of the fd range we will use to store the fds we pass to
+         * the process (or which are closed only during execve). When using fexecve() on a script we'll need
+         * to disable cloexec and the fd will be visible to the application. Let's move the fd very high up.
+         * Bash uses 255, let's do the same. */
+
+        int threshold = MAX(3 + (int) *n_fds, 255);
```

Comment:
maybe use FD_SETSIZE-1? i.e. 1023? i.e. the largest fd that select()-using programs can deal with?